### PR TITLE
Adds support for S3 feed

### PR DESCRIPTION
**Pull request by egorpavlikhin**
### Background

Adds support for basic S3 feed with supplied credentials. This package feed expects packages in the format of `{bucketName}/{packageName}` and will attempt to find matching files in the bucket by going through the list of know extensions (starting with `zip`). Having access to an S3 feed will enable serverless deployment scenarios, which are typically uploaded to an S3 bucket as a single file.

Related Server PR https://github.com/OctopusDeploy/OctopusDeploy/pull/11606

### How to review

I have done smoke testing and everything appears to work correctly. The one thing that might be surprising is the packageId format, but on the other hand we use the same format for github and docker feeds, so maybe that's ok.

I've only done a happy path test, as it's not clear what exactly needs testing, as the feed mostly calls to AWS SDK and we probably don't need to test that the SDK works.

### CODE REVIEW

File: source/Calamari.Shared/Calamari.Shared.csproj [hunk 1]
```diff
@@ -23,10 +23,7 @@
     <RootNamespace>Calamari</RootNamespace>
     <LangVersion>8</LangVersion>
     <Nullable>enable</Nullable>
-  </PropertyGroup>
-
-  <PropertyGroup Condition="!$([MSBuild]::IsOSUnixLike())">
-    <TargetFrameworks>net452;netstandard2.1;net40</TargetFrameworks>
+    <TargetFrameworks>net40;net452;netstandard2.1</TargetFrameworks>
```
Comment (andrewabest) [inline source/Calamari.Shared/Calamari.Shared.csproj:26]: Does the latter `TargetFramework` declaration overwrite this one if the condition is correct?

File: source/Calamari.Shared/Integration/Packages/Download/S3PackageDownloader.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                                                           string? feedUsername,
+                                                           string? feedPassword,
+                                                           bool forcePackageDownload,
+                                                           int maxDownloadAttempts,
+                                                           TimeSpan downloadAttemptBackoff)
+        {
+            var (bucketName, prefix) = GetBucketAndKey(packageId);
+            if (string.IsNullOrWhiteSpace(bucketName) || string.IsNullOrWhiteSpace(prefix))
+            {
+                throw new InvalidOperationException("Invalid PackageId for S3 feed. Expecting format `<bucketName>/<packageId>`");
```
Comment (andrewabest) [inline source/Calamari.Shared/Integration/Packages/Download/S3PackageDownloader.cs:62]: Should we put what was received in the exception too? i.e. `received {packageId}, expected format ...`

File: source/Calamari.Shared/Calamari.Shared.csproj (hunk 1 above)
Comment (egorpavlikhin) [reply source/Calamari.Shared/Calamari.Shared.csproj]: This is just an artifact of manual testing, I'll revert.

**Review by andrewabest — APPROVED** [review]
All looks good, just a couple of small comments.

**Pull request merged**